packages/neon-core/src/sc/manifest/ContractPermission.ts

```diff
@@ -0,0 +1,24 @@
+import { ContractPermissionDescriptor } from "./ContractPermissionDescriptor";
+
+export interface ContractPermissionLike {
+  contract: string;
+  methods: "*" | string[];
+}
+
+export class ContractPermission {
+  public contract: ContractPermissionDescriptor;
+  public methods: "*" | string[];
+
+  public constructor(obj: Partial<ContractPermissionLike>) {
+    const { contract = "*", methods = "*" } = obj;
+    this.contract = new ContractPermissionDescriptor(contract);
```

Comment:
> we might want to change the name of this field to `scope` if we d

I try to keep the consistence between the real manifest and the name of props in Manifest class. If we change it, user may be confused. (In manifest json got from rpc `getcontractstate`, it's called contract.) 

> Looks like we actually can just compress the descriptor logic into this class since its just simple validation logic.

Seems good. Except that we need the `isHash` `isGroup` interfaces, and it should belong to `contract` instead of the whole permission.
```ts
     const { contract, methods } = permission;
     if (contract.isHash) {
        if (contract.descriptor! !== manifest.hash) {
          return false;
        }
      } else if (contract.isGroup) {
        if (
          manifest.groups.every(group => group.pubKey !== contract.descriptor)
        ) {
          return false;
        }
      }
```

Reply:
`contract` kept, code compressed